### DISCUSSION TITLE
Multi-tileset Layers

### DISCUSSION THREAD

**Discussion by DelinWorks** [axmolengine/axmol#1132] [General]
@aismann (#617) @balancedwolf was trying to add support for multi-tileset layers but didn't have a response from a long time ago and is likely abandoned the PR.

I will be working on it. my way around it will be to split `TMXLayer` into `TMXSubLayer` a layer can have one or more sub-Layers, the only thing that will have a render command is a `TMXSubLayer` each sub-Layer will have it's own tileset, this implementation will be hidden from the user using your typical `TMXLayer`, when a layer is visited it will render all of it's sub-Layers. I don't know if this is the most efficient way to solve this problem, also the animation system has to be rewritten, it's very inefficient. the animation would be implemented in such a way that it changes all tiles with the same animation all at once, rather than calling `SetupTiles` and `setTileGIDAt` each frame for each tile which is quite expensive!

Any suggestions before I start would be awesome!

**Comment by aismann** [comment]
@DelinWorks 
Let its start.
